src/app/config/confparse.h

```diff
@@ -34,10 +34,36 @@ typedef struct config_deprecation_t {
  * you can abbreviate <b>tok</b>s as <b>tok</b>". */
 #define PLURAL(tok) { #tok, #tok "s", 0, 0 }
 
-/** Type of a callback to validate whether a given configuration is
- * well-formed and consistent. See options_trial_assign() for documentation
- * of arguments. */
-typedef int (*validate_fn_t)(const void*, void*, char**);
+/**
+ * Helper function for configuration formats: used to derive fields that
+ * can be detrmined before verifications.  Only this function and
+ * post_derive_fn_t are allowed to change the configuration object.
```

Comment:
Please document the arguments and return values for all these callbacks.

These void pointers make me uncomfortable, are you sure we can't replace them with some more specific types? If not, how do we ensure that each callback checks the magic values?